lib/bibliothecary/analyser.rb

```diff
@@ -26,10 +26,15 @@ def mapping_entry_match?(regex, details, filename, contents)
       def parse_file(filename, contents)
         mapping.each do |regex, details|
           if mapping_entry_match?(regex, details, filename, contents)
-            return send(details[:parser], contents)
+            begin
+              return send(details[:parser], contents)
+            rescue Exception => e # default is StandardError but C bindings throw Exceptions
+              # the C xml parser also puts a newline at the end of the message
+              raise Bibliothecary::FileParsingError.new(e.message.strip, filename)
+            end
           end
         end
-        return []
+        raise Bibliothecary::FileParsingError.new("No parser for this file type", filename)
```

Comment:
How do we get into this state? Shouldn't we only be calling `parse_file` if we matched?

Reply:
It happens if you don't check for match first. (The PR fixes one such bug in bibliothecary itself.) parse_file is "public" (I guess - I mean, who knows if anyone's using it directly)

Reply:
👍